akka-cluster/src/multi-jvm/scala/akka/cluster/DowningWhenOtherHasQuarantinedThisActorSystemSpec.scala

```diff
@@ -0,0 +1,126 @@
+        # speed up decision
+        akka.cluster.split-brain-resolver.stable-after = 5s
+        """)))
+
+  // exaggerate the timing issue by ,making the second node decide slower
+  // this is to more consistently repeat the scenario where the other side completes downing
+  // while the isolated part still has not made a decision and then see quarantined connections from the other nodes
+  nodeConfig(second)(ConfigFactory.parseString("akka.cluster.split-brain-resolver.stable-after = 15s"))
+
+  testTransport(on = true)
+}
+
+class DowningWhenOtherHasQuarantinedThisActorSystemMultiJvmNode1
+    extends DowningWhenOtherHasQuarantinedThisActorSystemSpec
+class DowningWhenOtherHasQuarantinedThisActorSystemMultiJvmNode2
+    extends DowningWhenOtherHasQuarantinedThisActorSystemSpec
+class DowningWhenOtherHasQuarantinedThisActorSystemMultiJvmNode3
+    extends DowningWhenOtherHasQuarantinedThisActorSystemSpec
+
+abstract class DowningWhenOtherHasQuarantinedThisActorSystemSpec
+    extends MultiNodeSpec(DowningWhenOtherHasQuarantinedThisActorSystemSpec)
+    with MultiNodeClusterSpec {
+  import DowningWhenOtherHasQuarantinedThisActorSystemSpec._
+
+  "Cluster node downed by other" must {
+
+    if (!ArterySettings(system.settings.config.getConfig("akka.remote.artery")).Enabled) {
+      // this feature only works in Artery, because classic remoting will not accept connections from
+      // a quarantined node, and that is too high risk of introducing regressions if changing that
+      pending
+    }
+
+    "join cluster" taggedAs LongRunningTest in {
+      awaitClusterUp(first, second, third)
+      enterBarrier("after-1")
+    }
+
+    "down itself" taggedAs LongRunningTest in {
+      runOn(first) {
+        testConductor.blackhole(first, second, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.blackhole(third, second, ThrottlerTransportAdapter.Direction.Both).await
+      }
+      enterBarrier("blackhole")
+
+      within(15.seconds) {
+        runOn(first) {
+          awaitAssert {
+            cluster.state.unreachable.map(_.address) should ===(Set(address(second)))
+          }
+          awaitAssert {
+            // second downed and removed
+            cluster.state.members.map(_.address) should ===(Set(address(first), address(third)))
+          }
+        }
+        runOn(second) {
+          awaitAssert {
+            cluster.state.unreachable.map(_.address) should ===(Set(address(first), address(third)))
+          }
+        }
+      }
+      enterBarrier("down-second")
+
+      runOn(first) {
+        testConductor.passThrough(first, second, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.passThrough(third, second, ThrottlerTransportAdapter.Direction.Both).await
+      }
+      enterBarrier("pass-through")
+
+      runOn(second) {
+        // shutting down itself
+        awaitCond(cluster.isTerminated)
```

Comment:
this is using the default timeout, which I guess is 3 seconds here. Maybe good to use an explicit timeout to clarify that it is downing itself because of the ThisActorSystemQuarantinedEvent and not by the ordinary SBR (15 seconds)